use-cases/NVIDIA-GPU-passthrough-and-Kata.md

```diff
@@ -0,0 +1,290 @@
+```
+machine_type = "q35"
+
+hotplug_vfio_on_root_bus = true
+pcie_root_port = 1
+```
+
+## Build Kata Containers kernel with GPU support
+The default guest kernel installed with Kata Containers does not provide GPU support. To use a Nvidia GPU with Kata Containers, you need to build a kernel with the necessary GPU support.
+The following kernel config options need to be enabled:
+```
+# Support PCI/PCIe device hotplug (Required for large BARs device)
+CONFIG_HOTPLUG_PCI_PCIE=y
+CONFIG_HOTPLUG_PCI_SHPC=y
+
+# Enable the MMIO access method for PCIe devices (Required for large BARs device)
+CONFIG_PCI_MMCONFIG=y
+
+# Support for loading modules (Required for load Nvidia drivers)
+CONFIG_MODULES=y
+CONFIG_MODULE_UNLOAD=y
+```
+
+The following kernel config options need to be disabled:
+```
+# Disable Open Source Nvidia driver nouveau
+# It conflicts with Nvidia official driver
+CONFIG_DRM_NOUVEAU=n
+```
+> **Note**: `CONFIG_DRM_NOUVEAU` is normally disabled by default. It is worth checking that it is not enabled in your kernel configuration to prevent any conflicts.
+
+
+Build the Kata Containers kernel with the previous config options, using the instructions described in [Building Kata Containers kernel](https://github.com/kata-containers/packaging/tree/master/kernel). For further details on building and installing guest kernels, see [the developer guide](https://github.com/kata-containers/documentation/blob/master/Developer-Guide.md#install-guest-kernel-images).
+
+There is an easy way to build a guest kernel that supports Nvidia GPU ([related PR](https://github.com/kata-containers/packaging/pull/938)):
+```
+## build guest kernel with http://github.com/kata-containers/packaging
+
+# prepare (download guest kernel source, generate .config)
+$ ./build-kernel.sh -v 4.19.86 -g setup
+
+# build guest kernel
+$ ./build-kernel.sh -v 4.19.86 -g build
+
+# install guest kernel
+$ sudo -E ./build-kernel.sh -v 4.19.86 -g install
+/usr/share/kata-containers/vmlinux-gpu.container -> vmlinux-4.19.86-69-gpu
+/usr/share/kata-containers/vmlinuz-gpu.container -> vmlinuz-4.19.86-69-gpu
+```
+
+To build NVIDIA Driver in kata container, `kernel-devel` is required.  
+This is a way to generate rpm packages for `kernel-devel`:
+```
+$ cd kata-linux-4.19.86-68
+$ make rpm-pkg
+Output RPMs:
+~/rpmbuild/RPMS/x86_64/kernel-devel-4.19.86_gpu-1.x86_64.rpm
+```
+> **Note**:
+> - `kernel-devel` should be installed in kata container before run Nvidia driver installer.
+> - Run `make deb-pkg` to build the deb package.
+
+Before using the new guest kernel, please update the kernel parameters in `configuration.toml`.
+```
+kernel = "/usr/share/kata-containers/vmlinuz-gpu.container"
+```
+
+## NVIDIA GPU pass-through mode with Kata Containers
+Use the following steps to pass a Nvidia GPU device in pass-through mode with Kata:
+
+1. Find the Bus-Device-Function (BDF) for GPU device on host:
+```
```

Comment:
Could you indent these code blocks so they align with the text of the bullet, for example:

<pre>
1. Find the Bus-...
    ```
    $ sudo lspci
    ```
</pre>

Reply:
updated